### DISCUSSION TITLE
Creating a new file in a subdirectory from a link

### DISCUSSION THREAD

**Discussion by srouden** [nvim-neorg/neorg#948] [Q&A]
Write a link like this:
{:$/dir/file:}[Description]
where the directory 'dir' doesn't exist in the current workspace. 
When following the link the file opens in a new buffer and it can be edited as expected. When the file is saved there is an error that the file can't be created. Creating the directory manually and saving the file again solves the problem.

Is it something that I'm missing here? 
I'm a recent nvim convert and have previously used vim and vimwiki for a long time. Vimwiki creates missing directories in this situation.

**Comment by esquires** [comment]
When I have a link that doesn't exist, that often means the file got moved. Would it make sense to have an interface that looks like this? I'd probably choose (3) below more often than not.

```
Whoops, "$/dir/file" does not exist. How would you like to proceed?

1) fix manually
2) create the direction/file and jump to it
3) fix the link to one of (in case there is a file with the same name. This could also use fuzzy matching in case the file is similar.)
    a) $/dir/dir1/file
    b) $/dir/dir2/file
```

One issue with the above is when the norg workspace is massive so generating the list in (3) may take time in that case. It seems like there is possibly some sql caching work being done in anticipation of gtd/zettelkasten that could possibly be leveraged. I am not involved with GTD/zettelkasten though so that may not be a thing.

https://github.com/nvim-neorg/neorg/issues/745

**Reply by Anrock** [reply]
Looks pretty good. To mitigate lag from third option the actual list of fuzzy-found files can be generated on demand and, optionally, on next screen. So like
```
Whoops, "$/dir/file" does not exist. How would you like to proceed?

1) fix manually
2) create the direction/file and jump to it
3) fix the link to one of (in case there is a file with the same name)
    a) $/dir/dir1/file
    b) $/dir/dir2/file
4) Search for similar names...
```
Only when 4 is selected the actual fuzzy stuff happens. When it's done - either update the current message replacing 4) with list of links or open list of links in separate message. 

To be fair it kinda sounds like poorman telescope replacement. So maybe this could be a fallback with telescope as primary?

**Comment by jumpyjacko** [comment]
I figured out that there is a flag for the `:write` command, `:write ++p` (from [here](https://neovim.io/doc/user/editing.html#%2B%2Bp)) that creates the missing directories when you write. If you don't want to type `++p` every time you write you can use an autocommand. I added this snippet to my configs and it works when I write after navigating to a link with a missing directory by pressing enter.
```lua
    vim.api.nvim_create_autocmd({ "BufWritePre" }, {
      callback = function(event)
        if event.match:match("^%w%w+://") then
          return
        end
        local file = vim.loop.fs_realpath(event.match) or event.match
        vim.fn.mkdir(vim.fn.fnamemodify(file, ":p:h"), "p")
      end,
    })
```
